DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -41,15 +60,59 @@ public void Setup()
             );
             A.CallTo(() => registrationDataService.RegisterDelegate(failingRegistrationModel)).Returns("-1");
 
-            registrationService = new RegistrationService(registrationDataService, passwordDataService, emailService,
-                centresDataService);
+            registrationService = new RegistrationService
+            (
+                registrationDataService,
+                passwordDataService,
+                emailService,
+                centresDataService
+            );
+        }
+
+        [Test]
+        public void Registering_delegate_with_approved_IP_registers_delegate_as_approved()
+        {
+            // When
+            registrationService.RegisterDelegate(testRegistrationModel, "localhost", "123.456.789.100");
+
+            // Then
+            A.CallTo(() =>
+                    registrationDataService.RegisterDelegate(
+                        A<DelegateRegistrationModel>.That.Matches(d => d.Approved)))
+                .MustHaveHappened();

```

Comment:
Could we also check here that the "approved" which is returned by the registrationService.RegisterDelegate(testRegistrationModel, "localhost", "123.456.789.100"); 

is true?